L107-node-noop-start.md

```diff
@@ -0,0 +1,33 @@
+Node: Make `Server#start` a no-op
+----
+* Author(s): murgatroid99
+* Approver: wenbozhu
+* Status: In Review
+* Implemented in: Node.js
+* Last updated: 2023-10-02
+* Discussion at: https://groups.google.com/g/grpc-io/c/zcGg6ipiZMo
+
+## Abstract
+
+Remove the functionality of the `start` method in the `Server` class, and deprecate that method.
+
+## Background
+
+Currently, a `Server` object effectively has a two-phase startup sequence: first, the user calls `bindAsync` to bind a port, then they call `start` to start serving. Specifically, `bindAsync` puts the server into a state where it allows clients to establish HTTP/2 sessions on the specified port, but then immediately closes those sessions. `start` causes the server to stop closing those sessions, and to instead serve requests on them.
+
+This behavior is problematic because allowing clients to establish HTTP/2 sessions signals to the client that the server is available, and clients will continue trying to connect to the same server instead of trying other servers. In addition, clients do not back off when reconnecting after a connection that was successfully established is closed, so this behavior will cause clients continuously reconnect with no backoff.
```

Comment:
What's the reason it was implemented in the first place? May be there was use-case?

Reply:
This code was contributed by a third-party developer, and I didn't dig into it at the time. I think the primary objective at the time was to emulate the behavior of the wrapped library. In the wrapped library, `bindAsync` called `grpc_server_add_insecure_http2_port` or `grpc_server_add_secure_http2_port`, and `start` called `grpc_server_start`.